From a277fd504e6e8978bbb5e088cc45961e2a804694 Mon Sep 17 00:00:00 2001
From: Andrei Lebedev <lebdron@gmail.com>
Date: Sun, 17 Sep 2017 10:34:22 +0800
Subject: [PATCH] Fix protobuf_generate_grpc_cpp when package is not located in
 usr/local

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5a7fad..4205a7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,16 +203,17 @@ elseif("${gRPC_PROTOBUF_PROVIDER}" STREQUAL "package")
     if(TARGET protobuf::protoc)
       set(_gRPC_PROTOBUF_PROTOC protobuf::protoc)
       set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
+      get_target_property(PROTOBUF_WELLKNOWN_IMPORT_DIR protobuf::libprotoc INTERFACE_INCLUDE_DIRECTORIES)
     else()
       set(_gRPC_PROTOBUF_PROTOC ${PROTOBUF_PROTOC_EXECUTABLE})
       set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
+      set(PROTOBUF_WELLKNOWN_IMPORT_DIR ${PROTOBUF_INCLUDE_DIRS})
     endif()
     set(_gRPC_FIND_PROTOBUF "if(NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND)\n  find_package(Protobuf ${gRPC_PROTOBUF_PACKAGE_TYPE})\nendif()")
   endif()
   if(PROTOBUF_FOUND)
     include_directories(${PROTOBUF_INCLUDE_DIRS})
   endif()
-  set(PROTOBUF_WELLKNOWN_IMPORT_DIR /usr/local/include)
 endif()
 
 if("${gRPC_SSL_PROVIDER}" STREQUAL "module")
-- 
2.7.4

